src/tools/rest-backend/benchmarks/benchmark_kdbrest_mysql.cpp

```diff
@@ -0,0 +1,621 @@
+		entry.setTitle (rs->getString ("title").asStdString ());
+		entry.setDescription (rs->getString ("description").asStdString ());
+		entry.setUploadPlugin (rs->getString ("plugin").asStdString ());
+		model::ImportedConfig cfg = service::ConvertEngine::instance ().import (rs->getString ("configuration").asStdString (),
+											entry.getUploadPlugin (), entry);
+		auto subkeys = cfg.getKeySet ();
+		entry.addSubkeys (subkeys.begin (), subkeys.end ());
+		entry.setAuthor (rs->getString ("author").asStdString ());
+		entry.setViews (rs->getInt ("views"));
+		entry.setCreatedAt (rs->getInt ("createdat"));
+
+		result.push_back (entry);
+	}
+
+	return result;
+}
+
+void printEntryVector (std::vector<model::Entry> entries, int indent = 0)
+{
+	std::string spaces = "";
+	for (int i = 0; i < indent; i++)
+	{
+		spaces += " ";
+	}
+	std::cout << spaces << "-> Result:" << std::endl;
+	for (auto & entry : entries)
+	{
+		std::cout << spaces << spaces << "-> " << entry.getPublicName () << std::endl;
+	}
+}
+
+void benchmarkLookupSingleByKey (sql::Connection * con, int numUsers, int numEntriesPerUser, int numTagsPerEntry, int indexes)
+{
+	std::cout << "Benchmark: Lookup single entry by fully specified key\n"
+		  << "          (" << numUsers << " users à " << numEntriesPerUser << " entries with " << numTagsPerEntry << " tags each)"
+		  << std::endl;
+	std::cout << "==============================================================" << std::endl;
+
+	std::cout << "-> Refreshing database (clear & create tables)" << std::endl;
+
+	clearDatabase (con, indexes);
+
+	std::cout << "-> Creating test data (" << (numUsers + numUsers * numEntriesPerUser + numUsers * numEntriesPerUser * numTagsPerEntry)
+		  << " rows)" << std::endl;
+
+	prepareTestData (con, numUsers, numEntriesPerUser, numTagsPerEntry);
+
+	std::cout << "-> Executing benchmark:" << std::endl;
+
+	// decide what key to look for
+	int user_index = numUsers / 2;
+	int entry_index = numEntriesPerUser / 2;
+	std::string organization = "organization-" + std::to_string (entry_index);
+	std::string application = "application-" + std::to_string (entry_index);
+	std::string scope = "scope-" + std::to_string (entry_index);
+	std::string slug = "slug-" + std::to_string (entry_index) + std::string ("-user-") + std::to_string (user_index);
+
+	// timer start
+	Timer timer;
+	timer.start ();
+
+	// the stuff to benchmark
+	sql::PreparedStatement * pstmt;
+	sql::ResultSet * rs;
+	pstmt = con->prepareStatement (
+		"SELECT s.*,u.`name` AS `author` "
+		"FROM `snippets` AS s "
+		"INNER JOIN `users` AS u "
+		"ON s.`author_id` = u.`id` "
+		"WHERE `organization` = ? AND `application` = ? AND `scope` = ? AND `slug` = ?");
+	pstmt->setString (1, organization);
+	pstmt->setString (2, application);
+	pstmt->setString (3, scope);
+	pstmt->setString (4, slug);
+	rs = pstmt->executeQuery ();
+
+	// stop timer here
+	timer.stop ();
+
+	// print found entries
+	std::vector<model::Entry> entries = rsToEntryVector (rs);
```

Comment:
You are right, this is a mistake. Definitely have to change and re-test this. Thank you for the good catch!